curated_transformers/tokenizers/auto_tokenizer.py

```diff
@@ -75,36 +110,63 @@ def from_hf_hub(cls, *, name: str, revision: str = "main") -> TokenizerBase:
             The tokenizer.
         """
 
-        tokenizer_cls = _resolve_tokenizer_class(name, revision)
+        tokenizer_cls = _resolve_tokenizer_class_hf_hub(name, revision)
         # This cast is safe, because we only return tokenizers.
         return cast(
             TokenizerBase, tokenizer_cls.from_hf_hub(name=name, revision=revision)
         )
 
 
 def _get_tokenizer_class_from_config(
-    *, name: str, revision: str
+    tokenizer_config: Dict[str, Any]
 ) -> Optional[Type[FromHFHub]]:
     """
     Infer the tokenizer class from the tokenizer configuration.
 
-    :param name:
-        Model name.
+    :param tokenizer_config:
+        The tokenizer configuration.
     :param revision:
         Model revision.
     :returns:
         Inferred class.
     """
+    return HF_TOKENIZER_MAPPING.get(tokenizer_config.get("tokenizer_class", None), None)
 
-    try:
-        tokenizer_config = get_tokenizer_config(name=name, revision=revision)
-    except EntryNotFoundError:
-        return None
 
-    return HF_TOKENIZER_MAPPING.get(tokenizer_config.get("tokenizer_class", None), None)
+def _resolve_tokenizer_class_fsspec(
+    fs: AbstractFileSystem,
+    model_path: str,
+    fsspec_args: Optional[Dict[str, Any]] = None,
+) -> Type[FromHFHub]:
+    fsspec_args = {} if fsspec_args is None else fsspec_args
+    tokenizer_cls: Optional[Type[FromHFHub]] = None
+    if fs.exists(f"{model_path}/{TOKENIZER_JSON}", **fsspec_args):
+        return Tokenizer
 
+    if tokenizer_cls is None:
+        tokenizer_config = get_tokenizer_config_fsspec(
+            fs=fs, model_path=model_path, fsspec_args=fsspec_args
+        )
+        if tokenizer_config is not None:
+            tokenizer_cls = _get_tokenizer_class_from_config(tokenizer_config)
 
-def _resolve_tokenizer_class(name: str, revision: str) -> Type[FromHFHub]:
+    if tokenizer_cls is None:
+        try:
+            model_type = get_model_type_fsspec(
+                fs=fs, model_path=model_path, fsspec_args=fsspec_args
+            )
+        except:
```

Comment:
Can we restrict the exception to a specific class or does it have to catch-all?

Reply:
I changed the `get_model_type_fsspec` method to return `Optional` rather than raising.